File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<int> count_alpha(string st){
        vector<int> alpha(26);
        for(int i = 0; i < st.size(); i++){
            alpha[st[i] - 'a']++;
        }
        return alpha;
    }
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<vector<int>, vector<string>> m;
        for(auto str: strs){
            m[count_alpha(str)].emplace_back(str);
        }
        vector<vector<string>> ans;
        for(auto [_, v]: m){
            ans.emplace_back(v);
        }
        return ans;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dic = dict()
        for st in strs:
            sorted_st = ''.join(sorted(st))
            dic.setdefault(sorted_st, []).append(st)
        ret = []
        for values in dic.values():
            ret.append(values)
        return ret